### DISCUSSION TITLE
How to configure maxSockets ?

### DISCUSSION THREAD

**Discussion by goriunov** [nodejs/undici#833] [Q&A] [answered]
Hello everyone,

We are thinking to use undici as our underline http client, but all of our systems deployed to Azure, and we need to make sure we do not cause snat port exhaustion when doing outbound calls.

Some read up https://docs.microsoft.com/en-us/azure/app-service/troubleshoot-intermittent-outbound-connection-errors

Azure mS propose to use an agent like https://www.npmjs.com/package/agentkeepalive. Which has maxSockets and maxFreeSockets configurations (and timeouts). I was not able to find anything in the undici about this.

Does anyone knows how to do any of those configurations or similar (i am not sure if it is possible to pass agentkeepalive to undici)

Would really appreciate any help !!!

Thank you :)

**Comment by mcollina** [comment, marked as answer]
```js
import { setGlobalDispatcher, Agent, request } from 'undici'

setGlobalDispatcher(new Agent({ connections: 42 }))

await request(..)
```

**Reply by goriunovphl** [reply]
Thank you ! Is there any alternatives to maxFreeSockets on undici ?

**Reply by mcollina** [reply]
Because of the design of undici, that configuration value is not necessary.
